table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,389 @@
+import com.amazonaws.services.glue.model.AWSGlueException;
+import com.amazonaws.services.glue.model.Column;
+import com.amazonaws.services.glue.model.Database;
+import com.amazonaws.services.glue.model.EntityNotFoundException;
+import com.amazonaws.services.glue.model.GetDatabaseRequest;
+import com.amazonaws.services.glue.model.GetDatabaseResult;
+import com.amazonaws.services.glue.model.GetPartitionsRequest;
+import com.amazonaws.services.glue.model.GetTableRequest;
+import com.amazonaws.services.glue.model.GetTablesRequest;
+import com.amazonaws.services.glue.model.GetTablesResult;
+import com.amazonaws.services.glue.model.GlueEncryptionException;
+import com.amazonaws.services.glue.model.Partition;
+import com.amazonaws.services.glue.model.Table;
+import com.amazonaws.services.glue.model.ValidationException;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Glue database implementation.
+ */
+public class GlueDatabase implements UnderDatabase {
+  private static final Logger LOG = LoggerFactory.getLogger(GlueDatabase.class);
+
+  private final UdbContext mUdbContext;
+  private final AWSGlueAsync mGlueClient;
+  private final UdbConfiguration mGlueConfiguration;
+  /** the name of the glue db. */
+  private final String mGlueDbName;
+
+  /** the owner name of glue database, which is an fake place holder. */
+  private final String mOwnerName = "PUBLIC_OWNER";
+  private final alluxio.grpc.table.PrincipalType mOwnerType = alluxio.grpc.table.PrincipalType.ROLE;
+
+  @VisibleForTesting
+  protected GlueDatabase(UdbContext udbContext, UdbConfiguration glueConfig, String glueDbName) {
+    mUdbContext = udbContext;
+    mGlueConfiguration = glueConfig;
+    mGlueClient = createAsyncGlueClient(glueConfig);
+    mGlueDbName = glueDbName;
+  }
+
+  /**
+   * Create an instance of the Glue database UDB.
+   *
+   * @param udbContext the db context
+   * @param configuration the configuration
+   * @return the new instance
+   */
+  public static GlueDatabase create(UdbContext udbContext, UdbConfiguration configuration) {
+    String glueDbName = udbContext.getUdbDbName();
+    if (glueDbName == null || glueDbName.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Glue database name cannot be empty: " + glueDbName);
+    } else if (configuration.get(Property.GLUE_REGION) == null) {
+      throw new IllegalArgumentException("GlueUdb Error: AWS region cannot be empty.");
+    }
+
+    return new GlueDatabase(udbContext, configuration, glueDbName);
+  }
+
+  @Override
+  public UdbContext getUdbContext() {
+    return mUdbContext;
+  }
+
+  @Override
+  public DatabaseInfo getDatabaseInfo() throws IOException {
+    LOG.info("Getting Glue database information from database: " + mGlueDbName + ".");
```

Comment:
Is this log message necessary? Should it be debug level?

Reply:
Fixed it.